forte/processors/misc/remote_processor.py

```diff
@@ -67,14 +68,40 @@ def initialize(self, resources: Resources, configs: Config):
                 f"{response.status_code}: "
                 "Remote service not started or invalid endpoint configs."
             )
+        service_name: str = response.json()["name"]
+
+        if self.configs.do_validation:
+            # Validate service name
+            if service_name != self.configs.expected_name:
+                raise ProcessorConfigError(
+                    "Validation fail: Name Mismatch. "
```

Comment:
The error message is unspecific. Here is an great article about this (and in general how to write good APIs): https://blog.keras.io/category/essays.html